### ISSUE TITLE
Fix critical accessibility issues

### ISSUE THREAD

**Issue by thrbnhrtmnn** [deven-org/boiler#595]
#### Description / User Story

After #187 this is the task to fix the most critical accessibility issues. Preparing this task is part of #187 .

TIMEBOX: 1 iteration

----
#### Requirements
- [x] #187 

----
#### Acceptance Criteria
- [x] All accessibility tests are passing and there are no validations or incomplete tests for the components:
   - [x] TextButton > @seemann 
   - [x] IconButton > @seemann 
   - [x] Checkbox
   - [x] Radio > @kaikdi 
   - [x] RadioGroup > @kaikdi  
   - [x] Select > @kaikdi
   - [x] TextInput
   - [x] TextArea
   - [x] ToggleSwitch 
   - [x] Counter
   - [x] FormCaption
   - [x] FormCaptionGroup
   - [x] FormLabel
   - [x] ButtonGroup @kaikdi 
   - [x] ~NumberInput~

----
#### Background information
- ...

----
#### Incomplete tests from #187 
- ~TextButton: Elements must have sufficient color contrast (disabled=true & theme=light)~
- ~Checkbox: Form field must not have multiple label elements (hasLabel=true)~
- ~Radio: ARIA attributes must conform to valid values (disabled=true / hasError=true)~
- NumberInput: Elements must only use permitted ARIA attributes (if ariaLabel has a value)

----
#### Violations from #187
- ~IconButton: Elements must only use allowed ARIA attributes (disabled=true)~
- ~TextButton:~
   - ~id attribute value must be unique (all states & variants)~
   - ~ARIA commands must have an accessible name (loading=true)~
   - ~Elements must have sufficient color contrast (disabled=true & theme=light / variant=secondary / variant=encourage)~ > Follow up tasks were created: #660 & #661
- Tooltip (should be checked again after #461 ): <object> elements must have alternate text (all variants except toolTipArrow=hide)
- ~Checkbox:~
   - ~Form elements must have labels (hasLabel=false / checkInputId="" / hasLabel=true & label="")~
   - ~Elements must have sufficient color contrast (theme=dark & hasError=true / theme=dark & showHint=true)~ 
- NumberInput:
   - Buttons must have discernible text (readonly=false / unit=defined)
   - Form elements must have labels (placeholder="")
   - Elements must meet minimum color contrast ratio thresholds (theme=dark)
- ~Radio:~
   - ~Form elements must have labels (label="" / optionId="")~ > this is just using the component wrong, we should not invest too much effort in preventing users to do that, there will almost always be a way to break things
   - ~Elements must have sufficient color contrast (disabled=true / theme=dark & showHint=true / theme=dark & hasError=true)~
- ~RadioGroup:~
   - ~ARIA attributes must conform to valid values (hasError=true)~
   - ~Elements must have sufficient color contrast (theme=dark & hasError=true / theme=dark & showHint=true)~
- ~Select: Elements must have sufficient color contrast (~theme=dark & showHint=true /~ theme=dark & hasError=true)~ > error state contrast needs to be fixed by design, a follow up task will be created
- ~TextInput:~ 
   - ~Elements must have sufficient color contrast (~theme=dark & hasError=true /~ theme=dark & showHint=true) ~  > error state contrast needs to be fixed by design, a follow up task will be created
   - ~Form elements must have labels (placeholder="")~
- ~TextArea: Elements must have sufficient color contrast (theme=dark & hasError=true / theme=dark & showHint=true / theme=dark & showCounter=true & warningState=true / theme=dark & showCounter=true & errorState=true / showCounter=true & size=md)~
- ~ToggleSwitch: Elements must have sufficient color contrast (theme=dark & showHint=true)~
- ~FormCaption (currently still named FormHint): Elements must have sufficient color contrast (theme=dark)~
- ~FormCaptionGroup (currently still named FormInfo): Elements must have sufficient color contrast (theme=dark)~
- ~FormLabel:~
   - ~Elements must have sufficient color contrast (theme=dark & variant=error)~
   - ~id attribute value must be unique (all states & variants)~
- ~ButtonGroup:~
   - ~ARIA commands must have an accessible name (all states & variants)~
   - ~Elements must have sufficient color contrast~ this only happens due to the "labels" next to the component, should be fixed with documentation task

### Follow-Up Task
Some components received new errors while this issue was in progress, these issues will be fixed in a follow-up task. This includes the following components:
- Checkbox

Incomplete tests:
- RadioGroup: ARIA attributes must conform to valid values  (disabled=true / hasError=true)
- Counter: Elements must have sufficient color contrast (size=md)
- TextArea: Elements must have sufficient color contrast (readonly=true / hasError=true /hasCounter=true & size=md)
- ToggleSwitch: Form field must not have multiple label elements (all states & variants)
- FormCaption: Elements must only use permitted ARIA attributes (if ariaLabel has a value)
- FormCaptionGroup: Elements must only use permitted ARIA attributes (probably inherited from FormCaption if ariaLabel has a value)

Components that need to be checked as well after refactoring tasks:
- Tooltip
- TabBar
- SliderSingleValue
- SliderTwoValues
